wordpress-mobile/WordPress-iOS
# Refactor Jetpack connection screen for stats

This PR accomplishes 2 things:
1. It updates the existing (broken) approach of logging into .com so a jetpack-enabled .org site can use stats within WPiOS. 
2. As part of the iPad Refresh project, the Jetpack login UI now uses autolayout and stackviews avoiding previous `.frame` manipulation.  The new screens should render well on all devices as well as in various multitasking modes.

The old UI+logic from `JetpackSettingsViewController.h`/`JetpackSettingsViewController.m` was refactored into `JetpackLoginViewController.swift` with the corresponding UI housed in `JetpackLoginViewController.xib`. 

Note that the existing `JetpackService` class is no longer used because it did not behave correctly when 2FA was enabled. Instead, the `LoginFacade` class from our service layer is leveraged. 

<img width="375" alt="screen shot 2017-01-30 at 1 09 11 pm" src="https://cloud.githubusercontent.com/assets/154014/22437359/64435cb2-e6ed-11e6-8827-31d604ff7d44.png"><img width="375" alt="screen shot 2017-01-30 at 1 09 33 pm" src="https://cloud.githubusercontent.com/assets/154014/22437362/68116ae6-e6ed-11e6-8263-90fbf99da0c6.png">
<img width="266" alt="screen shot 2017-01-30 at 1 07 05 pm" src="https://cloud.githubusercontent.com/assets/154014/22437445/bd727ec6-e6ed-11e6-9586-55bece28d8ed.png"><img width="333" alt="screen shot 2017-01-30 at 1 07 42 pm" src="https://cloud.githubusercontent.com/assets/154014/22437447/c0e7822c-e6ed-11e6-83aa-89d798dd8249.png">

Fixes #6389 and #4521

**To test:**
Initial preconditions:
1. Log into a .org site with Jetpack installed using a userid from the .org site (not .com)
2. Tap on the Stats section the .org site you should be presented with a login form.

At this point, there are _lots_ of things to test:
* Try breaking the login form by using bad inputs
* Test the error handling
* Try using a .com login with 2FA AND without 2FA turned on
* Does the form dismiss properly when login is successful?
* What happens with the network connection is sketchy?
* Does the login form become hidden after rotating? (when inputting fields, the entire scrollview should scroll vertically only)
* Does killing app mid-login cause any issues after returning to the app?

Of course, this screen should render correctly on all device sizes and work as expected with a soft keyboard present or not. In addition be sure to test multitasking support on the iPad Pros.

Needs review: @astralbodies would you be so kind?
